Basic_viewer/doc/Basic_viewer/CGAL/Qt/Basic_viewer_qt.h

```diff
@@ -0,0 +1,256 @@
+
+  /// adds the given point in the viewer.
+  template <typename KPoint>
+  void add_point(const KPoint &p);
+
+  /// adds the given colored point in the viewer.
+  template <typename KPoint>
+  void add_point(const KPoint &p, const CGAL::IO::Color &acolor);
+
+  /// adds the given segment in the viewer.
+  template <typename KPoint>
+  void add_segment(const KPoint &p1, const KPoint &p2);
+
+  /// adds the given colored segment in the viewer.
+  template <typename KPoint>
+  void add_segment(const KPoint &p1, const KPoint &p2,
+                   const CGAL::IO::Color &acolor);
+
+  /// adds the given ray in the viewer: an half line starting from `p` and having `v` as direction.
+  template <typename KPoint, typename KVector>
+  void add_ray(const KPoint &p, const KVector &v);
+
+  /// adds the given colored ray in the viewer: an half line starting from `p` and having `v` as direction.
+  template <typename KPoint, typename KVector>
+  void add_ray(const KPoint &p, const KVector &v,
+               const CGAL::IO::Color &acolor);
+
+  /// adds the given line in the viewer, defined by `p` and `v` as direction.
+  template <typename KPoint, typename KVector>
+  void add_line(const KPoint &p, const KVector &v);
+
+  /// adds the given colored line in the viewer, defined by `p` and `v` as direction.
+  template <typename KPoint, typename KVector>
+  void add_line(const KPoint &p, const KVector &v,
+                const CGAL::IO::Color &acolor);
+
+  /// starts a new face.
+  void face_begin();
+
+  /// starts a new colored face.
+  void face_begin(const CGAL::IO::Color &acolor);
+
+  /// returns `true` iff a face is started.
+  bool is_a_face_started() const;
+
+  /// adds the given point in the current face.
+  /// @pre `is_a_face_started()`
+  template <typename KPoint> bool add_point_in_face(const KPoint &kp);
+
+  /// adds the given point in the current face, having the vertex normal.
+  /// @pre `is_a_face_started()`
+  template <typename KPoint, typename KVector>
+  bool add_point_in_face(const KPoint &kp, const KVector &p_normal);
+
+  /// ends the current face.
+  /// @pre `is_a_face_started()`
+  void face_end();
+
+  /// adds the given text at the given position in the viewer.
+  template <typename KPoint>
+  void add_text(const KPoint &kp, const char *txt);
+
+  /// adds the given text at the given position in the viewer.
+  template <typename KPoint>
+  void add_text(const KPoint &kp, const std::string &txt);
+
+  /// returns the graphics scene of the viewer.
+  Graphics_scene<BufferType>& get_graphics_scene();
+
+  /// returns the graphics scene of the viewer, const version.
+  const Graphics_scene<BufferType>& get_graphics_scene() const;
+
+  /// negates all normals of vertices and faces.
+  void negate_all_normals();
+
+  /// returns `true` if the data structure in in 2D, i.e., lies on a plane.
+  bool is_two_dimensional() const;
+
+  /// draws the viewer without recomputing all internal buffers.
```

Comment:
draws the scene ???

Reply:
This is a method of basic viewer; thus maybe it is better to keep "draws the viewer" ?